### ISSUE TITLE
Multichannel subscription gets stuck

### ISSUE THREAD

**Issue by indrekj** [slact/nchan#454]
I think this is related to #448 .

I have an endpoint with multiple channels.
```
        location = /publish {
          nchan_publisher;
          nchan_channel_id $arg_channel_id;
          nchan_message_buffer_length 50;
          nchan_message_timeout 1h;
          nchan_redis_pass redis_cluster;
        }

        location = / {
          nchan_pubsub;
          nchan_subscriber_channel_id $arg_channel_id;
          nchan_channel_id_split_delimiter ",";
          nchan_subscriber_last_message_id $arg_last_message_id;
          nchan_publisher_channel_id ackchannel;
          nchan_redis_pass redis_cluster;
          nchan_max_channel_id_length 9434;
        }
```

I noticed at some point messages aren't coming through. `curl -i ..` with multiple channels just doesn't return anything. However `curl -i` against only one channel works fine. The other channel still doesn't return anything.

I looked into what's happening in redis:
```
127.0.0.1:6379> keys "*6c7d7b42-e2b2-467b-b190-680ede27962e*"
1) "{channel:/6c7d7b42-e2b2-467b-b190-680ede27962e}:messages"
2) "{channel:/6c7d7b42-e2b2-467b-b190-680ede27962e}"
```
The channel is listed in redis.

Messages include:
```
127.0.0.1:6379> LRANGE "{channel:/6c7d7b42-e2b2-467b-b190-680ede27962e}:messages" 0 10
1) "1524242345:0"
```

and metadata includes:
```
127.0.0.1:6379> hgetall "{channel:/6c7d7b42-e2b2-467b-b190-680ede27962e}"
 1) "current_message"
 2) "1524242345:0"
 3) "time"
 4) "1524242345"
 5) "max_stored_messages"
 6) "50"
 7) "prev_message"
 8) "1524236462:0"
 9) "last_subscriber_id"
10) "29"
11) "subscribers"
12) "4"
13) "last_seen_subscriber"
14) "1524145395"
15) "fake_subscribers"
16) "42"
17) "last_seen_fake_subscriber"
18) "1524242456"
```

but interesting is that there's no message data. I see other channels have keys like: `"{channel:/93f27270-bf46-46a1-a856-3a167c743930}:msg:1524240697:0"` but this one doesn't.

nchan info:
```
total published messages: 4525
stored messages: 139
shared memory used: 96K
shared memory limit: 131072K
channels: 15
subscribers: 6
redis pending commands: 0
redis connected servers: 3
total interprocess alerts received: 4991371
interprocess alerts in transit: 0
interprocess queued alerts: 0
total interprocess send delay: 0
total interprocess receive delay: 0
nchan version: 1.1.14
```

Restarting nchan fixes the problem for some time.

**Comment by indrekj** [comment]
Disabling `nchan_message_timeout` did not seem to help.

**Comment by indrekj** [comment]
Downgrade to `1.1.7.2` seemed to help.

**Comment by slact** [comment]
This issue should now be fixed in master. Please test it if you can.

**Comment by slact** [comment]
This issue is fixed in version 1.2.0. Please reopen if you encounter it again.
